stl/inc/xstring

```diff
@@ -198,17 +232,40 @@ struct _WChar_traits { // char_traits for the char16_t-likes: char16_t, wchar_t,
 #endif // _HAS_CXX17
     }
 
-    static _Elem* move(_Out_writes_all_(_Count) _Elem* const _First1, _In_reads_(_Count) const _Elem* const _First2,
-        const size_t _Count) noexcept /* strengthened */ {
+    static _CONSTEXPR20 _Elem* move(_Out_writes_all_(_Count) _Elem* const _First1,
+        _In_reads_(_Count) const _Elem* const _First2, const size_t _Count) noexcept /* strengthened */ {
         // copy [_First2, _First2 + _Count) to [_First1, ...)
-        return reinterpret_cast<_Elem*>(
-            _CSTD wmemmove(reinterpret_cast<wchar_t*>(_First1), reinterpret_cast<const wchar_t*>(_First2), _Count));
+#ifdef __cpp_lib_is_constant_evaluated
+        if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            return reinterpret_cast<_Elem*>(
+                _CSTD wmemmove(reinterpret_cast<wchar_t*>(_First1), reinterpret_cast<const wchar_t*>(_First2), _Count));
+        }
+
+        for (size_t _Idx = 0; _Idx < _Count; ++_Idx) {
+            *(_First1 + _Idx) = *(_First2 + _Idx);
+        }
+
+        return static_cast<_Elem*>(_First1);
     }
 
-    static _Elem* assign(_Out_writes_all_(_Count) _Elem* const _First, size_t _Count, _Elem _Ch) noexcept
+    static _CONSTEXPR20 _Elem* assign(_Out_writes_all_(_Count) _Elem* const _First, size_t _Count, _Elem _Ch) noexcept
     /* strengthened */ {
         // assign _Count * _Ch to [_First, ...)
-        return reinterpret_cast<_Elem*>(_CSTD wmemset(reinterpret_cast<wchar_t*>(_First), _Ch, _Count));
+#ifdef __cpp_lib_is_constant_evaluated
+        if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            return reinterpret_cast<_Elem*>(_CSTD wmemset(reinterpret_cast<wchar_t*>(_First), _Ch, _Count));
+        }
+
+        _Elem* _Next = _First;
+        for (; 0 < _Count; --_Count, ++_Next) {

```

Comment:
`_Next` isn't used outside this loop, and eww Yoda conditional:
```suggestion
        for (_Elem* _Next = _First; _Count > 0; --_Count, ++_Next) {
```

Reply:
That is what you get for mindlessly copying code around